File: maple-engine-plugins/maple-spark/apps/maple-spark-data-calc/src/main/scala/org/xi/maple/datacalc/spark/transform/SqlTransform.scala
package org.xi.maple.datacalc.spark.transform

import org.apache.spark.sql.{Dataset, Row}
import org.xi.maple.common.util.VariableUtils
import org.xi.maple.datacalc.spark.api.MapleTransform
import org.xi.maple.datacalc.spark.model.transform.SqlTransformConfig

class SqlTransform extends MapleTransform[SqlTransformConfig] {

  override protected def getData(variables: java.util.Map[String, String]): Dataset[Row] = {
    val selectSql = VariableUtils.replaceVariables(config.getSql, variables)
    logger.info(s"Load data from query: $selectSql")
    spark.sql(selectSql)
  }
}

File: maple-engine-plugins/maple-flink/apps/maple-flink-data-calc/src/main/java/org/xi/maple/datacalc/flink/transform/SqlTransform.java
package org.xi.maple.datacalc.flink.transform;

import org.apache.flink.table.api.TableEnvironment;
import org.xi.maple.datacalc.flink.api.MapleTransform;
import org.xi.maple.datacalc.flink.model.transform.SqlTransformConfig;

public class SqlTransform extends MapleTransform<SqlTransformConfig> {

    public SqlTransform(TableEnvironment tableEnv) {
        super(tableEnv);
    }

    @Override
    public void define() {
        tableEnv.createTemporaryView(config.getResultTable(), tableEnv.sqlQuery(config.getSql()));
    }
}
